manager/state/raft/membership/cluster.go

```diff
@@ -229,8 +249,7 @@ func (c *Cluster) CanRemoveMember(from uint64, id uint64) bool {
 			continue
 		}
 
-		connState, err := m.Conn.State()
-		if err == nil && connState == grpc.Ready {
+		if err := m.HealthCheck(); err == nil {
```

Comment:
If there are several members, and some of them are unreachable, that could block `CanRemoveMember` for a significant amount of time.

That doesn't seem terrible (I don't think it blocks actual raft replication), but I wonder if we can avoid it.

I definitely agree this should take a context.